src/encoder/plugins/OpusEncoderPlugin.cxx

```diff
@@ -183,24 +190,25 @@ OpusEncoder::~OpusEncoder()
 void
 OpusEncoder::DoEncode(bool eos)
 {
-	assert(buffer_position == buffer_size);
+	if (!eos)
+		assert(buffer_position == buffer_size);
 
 	opus_int32 result =
 		audio_format.format == SampleFormat::S16
 		? opus_encode(enc,
-			      (const opus_int16 *)buffer,
-			      buffer_frames,
-			      buffer2,
-			      sizeof(buffer2))
+		              (const opus_int16 *)buffer,
+		              buffer_frames,
+		              buffer2,
+		              sizeof(buffer2))
 		: opus_encode_float(enc,
-				    (const float *)buffer,
-				    buffer_frames,
-				    buffer2,
-				    sizeof(buffer2));
+		                    (const float *)buffer,
+		                    buffer_frames,
+		                    buffer2,
+		                    sizeof(buffer2));
 	if (result < 0)
 		throw std::runtime_error("Opus encoder error");
 
-	granulepos += buffer_frames;
+	granulepos += buffer_position / frame_size;
```

Comment:
What is this change about?

Reply:
on final stream packet (one with eos set), we are allowed to have granulepos less than packet actually contains. decoder should take smaller granulepos into account and discard unused data. this is for stream cutting - we need to push final packet to inject tags, but we don't have enough data for whole packet. therefore we encode full packet, but specify lesser granulepos.